test/extended/openstack/loadbalancers.go

```diff
@@ -412,6 +417,61 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack][lb][Serial] The O
 			o.Expect(podName).Should(o.BeElementOf(podNames))
 			e2e.Logf("Pod %s successfully accessed through svc loadbalancer on ip %s and port %d", podName, fip.FloatingIP, svcPort)
 		})
+
+		// https://issues.redhat.com/browse/OSASINFRA-2412
+		g.It(fmt.Sprintf("should create a TCP %s LoadBalancer when LoadBalancerService ingressController is created on Openshift", lbProviderUnderTest), func() {
+
+			skipIfNotLbProvider(lbProviderUnderTest, cloudProviderConfig)
+
+			g.By("Creating ingresscontroller for sharding the OCP in-built canary service")
+			name := "canary-sharding"
+			domain := "sharding-test.apps.shiftstack.com"
+			shardIngressCtrl, err := deployLbIngressController(oc, 10*time.Minute, name, domain, map[string]string{"ingress.openshift.io/canary": "canary_controller"})
+			defer func() {
+				err := oc.AdminOperatorClient().OperatorV1().IngressControllers(shardIngressCtrl.Namespace).Delete(context.Background(), shardIngressCtrl.Name, metav1.DeleteOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred(), "error destroying ingress-controller created during the test")
+			}()
+			o.Expect(err).NotTo(o.HaveOccurred(), "new ingresscontroller did not rollout")
+
+			g.By("Checks from openshift perspective")
+			svc, err := oc.AdminKubeClient().CoreV1().Services("openshift-ingress").Get(context.Background(), "router-"+name, metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred(), "router-%q not found in openshift-ingress namespace", name)
+			o.Expect(svc.Spec.Type).Should(o.Equal(v1.ServiceTypeLoadBalancer), "Unexpected svc Type: %q", svc.Spec.Type)
+			loadBalancerId := svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]
+			o.Expect(loadBalancerId).ShouldNot(o.BeEmpty(), "load-balancer-id annotation missing")
+			o.Expect(svc.Status.LoadBalancer.Ingress).ShouldNot(o.BeEmpty(), "svc.Status.LoadBalancer.Ingress should not be empty")
+			svcIp := svc.Status.LoadBalancer.Ingress[0].IP
+			o.Expect(svcIp).ShouldNot(o.BeEmpty(), "FIP missing on svc Status")
+			o.Expect(svc.Spec.ExternalTrafficPolicy).Should(o.Equal(v1.ServiceExternalTrafficPolicyTypeLocal),
+				"Unexpected ExternalTrafficPolicy on svc specs")
```

Comment:
Feels like we're testing what should be covered already in origin here, but I don't see it as an important issue.

Reply:
This test includes also openstack checks. Origin is only openshift.